phoenix-core-server/src/main/java/org/apache/phoenix/replication/log/LogFile.java

```diff
@@ -0,0 +1,579 @@
+         * Gets the number of columns included in this record's mutation data.
+         * @return The count of columns.
+         */
+        int getColumnCount();
+
+        /**
+         * Gets an iterable view of the column name + column qualifier to value mappings for this
+         * record. The underlying collection should not be modified via this iterable.
+         * @return An iterable of Map entries representing column names and their values.
+         */
+        Iterable<Map.Entry<byte[], Map<byte[], byte[]>>> getColumnValues();
+
+        /**
+         * Adds a column+qualifier and its corresponding value to this record.
+         * @param family The family as a byte array.
+         * @param qualifier The qualifier as a byte array.
+         * @param value The value of the column as a byte array.
+         * @return This Record instance for chaining.
+         */
+        Record addColumnValue(byte[] family, byte[] qualifier, byte[] value);
+
+        /**
+         * Removes all column name/value pairs currently stored in this record.
+         * @return This Record instance for chaining.
+         */
+        Record clearColumnValues();
+
+        /**
+         * Gets the total serialized length of this record in bytes, including any length prefixes
+         * used by the codec. This value should be set by the codec after writing or reading.
+         * @return The total serialized length of the record.
+         */
+        int getSerializedLength();
+
+        /**
+         * Sets the total serialized length of this record. This is typically called by the codec.
+         * @param serializedLength The total serialized length in bytes.
+         * @return This Record instance for chaining.
+         */
+        Record setSerializedLength(int serializedLength);
+
+        // Helper methods for converting between Records and HBase Mutations
+
+        /**
+         * Builds a Record from an HBase Mutation. Assumes the mutation contains cells
+         * relevant to a single logical operation at a specific timestamp.
+         * @param schemaObjectName The schema object name
+         * @param commitId The commit identifier
+         * @param mutation The HBase Mutation (Put or Delete).
+         * @return The corresponding Record instance.
+         * @throws IllegalArgumentException if the mutation type is unsupported.
+         */
+        static Record fromHBaseMutation(String schemaObjectName, long commitId, Mutation mutation) {
+            return LogFileRecord.fromHBaseMutation(schemaObjectName, mutation, commitId);
+        }
+
+       /**
+         * Builds an HBase mutation, given a Record.
+         * @param mutation The Record instance.
+         * @return The HBase Mutation
+         */
+        static Mutation toHBaseMutation(Record record) {
+            if (record == null) {
+                return null;
+            }
+            return LogFileRecord.toHBaseMutation(record);
+        }
+    }
+
+    /**
```

Comment:
When we simplify the Record interface as in my previous comment, we do not need redefine the MutationType, I think.